lib/ssl/src/ssl.erl

```diff
@@ -2750,16 +2750,17 @@ handle_supported_groups_option(Value, _Version0) when is_list(Value) ->
                      | closed
                      | {tls_alert, {_, Description :: string()}}
                      | {options, Options :: term()}
-                     | {options, {socket_options, Option :: term()}}
-                     | {options, {socket_options, Option :: term(), Error}}
+                     | {options, {socket_options, Option}}
+                     | {options, {socket_options, Option, Error}}
```

Comment:
I think we should use filename() instead of string().  And for the rest why do we need the name? For this internal function I think it will be enough with term(), documentation is for internal documentation of the structure and not something for generating API documentation.